### PR TITLE
League one teleop and autonomous

### PR DESCRIPTION
**Pull request by valsei**
This code implements all of the teleop driver controls and features relevant to the state of the robot as of league one (push bot):
- mecanum style driving
- slowmode (drive/turn speed)
- turn 90 degrees left/right buttons
- spinning and reversing intake motor
It also implements a basic drive inches autonomous, but note that it also includes an unused turn method that was in development.

### CODE REVIEW

File: Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/MecanumDrive.java
Comment (ThatSmartGuy21) [inline Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/MecanumDrive.java:1]: stolen from rr

File: Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/Utilities.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     * keeps angles between [-180,180] so that it does not exceed possible imu readings
+     * tested in https://www.desmos.com/calculator/igccxromri
+     * @param angle angle to limit (degrees)
+     * @return equivalent angle between -180 and 180
+     */
+    public static double limitAngle(double angle) {
+        return gooderMod((angle + 180.0), 360.0) - 180.0;
+    }
+    // does modulus similar to desmos mod function, normal java % is different and doesn't work
+    public static double gooderMod(double a, double b) {
```
Comment (ThatSmartGuy21) [inline Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/Utilities.java:14]: Name could be more descriptive. How is this different from the modulus operator (%)?

File: Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    public enum AutoAlliance {
+        BLUE,
+        RED
+    }
+    public enum AutoType {
+        SHORT,
+        LONG
+    }
+
+    public void doAutoDriveInches(AutoAlliance autoAlliance, AutoType autoType) throws InterruptedException {
```
Comment (ThatSmartGuy21) [inline Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java:26]: Name does not properly describe the action performed.

File: Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+                driveInches(-3 * strafeSignFlip, 0); // strafe away from the wall 3 inches
+                driveInches(0, 24 * 4); // drive forward to parking spot
+
+                break;
+        }
+
+    }
+
+
+    static final double TICKS_PER_REVOLUTION = 537.7;
```
Comment (ThatSmartGuy21) [inline Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java:56]: Put into constants class please :)

File: Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        BR.setTargetPosition((int) targetBR);
+
+        // Turn On RUN_TO_POSITION
+        FL.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+        FR.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+        BL.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+        BR.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+
+        // reset the timeout time and start motion.
+        //runtime.reset();
```
Comment (ThatSmartGuy21) [inline Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java:94]: Please remove commented out code please :))

File: Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        BR.setPower(ROBOT_SPEED);
+
+        // keep looping while we are still active, and there is time left, and both motors are running.
+        // Note: We use (isBusy() || isBusy()) in the loop test, which means that when EITHER motor hits
+        // its target position, the motion will stop.  This is "safer" in the event that the robot will
+        // always end the motion as soon as possible.
+        // However, if you require that BOTH motors have finished their moves before the robot continues
+        // onto the next step, use (isBusy() && isBusy()) in the loop test.
+        while (opModeIsActive() &&
+                //(runtime.seconds() < 30) &&
```
Comment (ThatSmartGuy21) [inline Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java:107]: commented out code again :)))

File: Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+        stopDriving();
+
+        // Turn off RUN_TO_POSITION
+        FL.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        FR.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        BL.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        BR.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+    }
+
+    public static double MIN_HEADING_ACCURACY = 5.0; // degrees off from target
```
Comment (ThatSmartGuy21) [inline Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java:121]: Put in constants class please :))))

File: Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_Auto_Test.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
+package org.firstinspires.ftc.team6220_CENTERSTAGE.autoClasses;
+
+import com.acmerobotics.dashboard.config.Config;
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+
+@Config
+@Autonomous(name="League1_Auto_Test", group ="amogus2")
+public class League1_Auto_Test extends League1_AutoFramework {
```
Comment (ThatSmartGuy21) [inline Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_Auto_Test.java:8]: Please disable this opmode :))))))

File: Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/teleOpClasses/League1_TeleOp.java [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+    double drivePowerY = 0.0;
+    double turnPower = 0.0;
+    double intakePower = 0.0;
+
+    // holds heading from imu read which is done in roadrunner's mecanum drive class for us
+    double currentHeading = 0.0;
+    double targetHeading = 0.0;
+
+    // constants
+    final double TURN_STICK_DEADZONE = 0.01;
```
Comment (ThatSmartGuy21) [inline Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/teleOpClasses/League1_TeleOp.java:64]: Constants class :)))))))

**Review by ThatSmartGuy21 — CHANGES_REQUESTED** [review]
Added comments, please fix :))))))))

File: Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java (hunk 3 above)
Comment (tmicro) [reply Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java:56]: The TICKS_PER_REVOLUTION constant value is not correct.

**Review by tmicro — APPROVED** [review]
Approved.

**Review by ThatSmartGuy21 — APPROVED** [review]
heck yeah

**Pull request merged**